### PR TITLE
transform and rater tests

### PR DESCRIPTION
**Pull request by EdTeng1**

### CODE REVIEW

**Comment by CambioML** [comment]
@DavidHHShao @SayaZhang Please help review.

File: tests/flow/transform/test_azure_openai.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    @patch('uniflow.flow.transform.transform_azure_openai_flow.LmModel')
+    @patch('uniflow.flow.transform.transform_azure_openai_flow.JsonLmModel')
+    def setUp(self, mock_json_model, mock_lm_model, mock_model_op):
+        self.mock_json_model = mock_json_model
+        self.mock_lm_model = mock_lm_model
+        self.mock_model_op = mock_model_op
+        self.prompt_template = PromptTemplate(instruction="instruction", few_shot_prompt=[{}])
+        self.model_config1 = {"response_format": {"type": "json_object"}}
+        self.model_config2 = {"response_format": {"type": "other"}}
+        self.azure_flow1 = AzureOpenAIModelFlow(self.prompt_template, self.model_config1)
```
Comment (SayaZhang) [inline tests/flow/transform/test_azure_openai.py:20]: shall we add test cases of prompt_templete=none and model_config=none?

File: tests/flow/transform/test_azure_openai.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        self.mock_json_model.assert_called_once_with(prompt_template=self.prompt_template, model_config=self.model_config1)
+        self.mock_model_op.assert_called()
+    
+    def test_not_json_init(self):
+        self.mock_lm_model.assert_called_once_with(prompt_template=self.prompt_template, model_config=self.model_config2)
+        self.mock_model_op.assert_called()
+
+    def test_run(self):
+        node1 = Node(name="node1", value_dict={"a": 1})
+        result = self.azure_flow1.run(node1)
```
Comment (SayaZhang) [inline tests/flow/transform/test_azure_openai.py:33]: nit: shall we add a new line here?
```
def test_call_with_empty_node(self, mock_read_file):
        # arrange
        nodes = []

        # act
        output_nodes = self.extract_txt_op(nodes)

        # assert
        mock_read_file.assert_not_called()
        self.assertEqual(len(output_nodes), 0)

```

File: tests/flow/transform/test_azure_openai.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        self.mock_model_op.return_value.assert_called_once_with(node1)
+        expected_result = self.mock_model_op.return_value(node1)
+        self.assertEqual(result, expected_result)
```
Comment (SayaZhang) [inline tests/flow/transform/test_azure_openai.py:36]: nit: same above, shall we add a new line here to indicate that this is an assert block

File: tests/flow/transform/test_google_flow.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+class TestGoogleModelFlow(unittest.TestCase):
+    @patch('uniflow.flow.transform.transform_google_flow.LmModel')
+    @patch('uniflow.flow.transform.transform_google_flow.ModelOp')
+    def setUp(self, mock_model_op, mock_lm_model):
+        self.mock_model_op = mock_model_op
+        self.mock_lm_model = mock_lm_model
+        self.mock_lm_model.return_value = MagicMock()
+        self.prompt_template = PromptTemplate(instruction="instruction", few_shot_prompt=[{}])
+        self.model_config = {"response_format": {"type": "json_object"}}
+        self.google_flow = GoogleModelFlow(self.prompt_template, self.model_config)
```
Comment (SayaZhang) [inline tests/flow/transform/test_google_flow.py:17]: same above, we should add two more unit tests to test with prompt_template and mode_config are none

Comment (DavidHHShao) [reply tests/flow/transform/test_google_flow.py:17]: we can control using python type hints to enforce the none check in the constructor.  Let me figure out how to set it up

**Comment by DavidHHShao** [comment]
Looks like most the tests are for object init with external mocks, which may not be valuable.  Let me consolidate some thoughts and get back with some suggestions
